x-pack/plugins/fleet/public/applications/fleet/hooks/use_apm_service_href.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Overwrite } from 'utility-types';
+import useAsync from 'react-use/lib/useAsync';
+
+import { useLocator } from '../../../hooks/use_locator';
+import type { DataStream } from '../types';
+import { LOCATORS_IDS } from '../constants';
+
+type APMDataStream = Overwrite<
+  DataStream,
+  { package: 'apm'; serviceDetails: NonNullable<DataStream['serviceDetails']> }
+>;
+
+export const isAPMIntegration = (datastream: DataStream): datastream is APMDataStream =>
+  Boolean(datastream.package === 'apm' && datastream.serviceDetails);
+
+export const useAPMServiceHref = (datastream: DataStream) => {
+  const apmLocator = useLocator(LOCATORS_IDS.APM_LOCATOR);
+
+  const { error, loading, value } = useAsync(() => {
+    if (!isAPMIntegration(datastream)) return Promise.resolve();
+
+    const { serviceName, environment } = datastream.serviceDetails;
+
+    return apmLocator!.getUrl({
+      serviceName,
+      serviceOverviewTab: datastream.type,
+      query: {
+        environment,
+      },
+    });
+  });
+
+  return { isReady: !error && !loading, href: value };
```

Comment:
I'm not sure if `isReady` should be false if `error` is non-null. I think it can be easily confused with loading state.